README.md

```diff
@@ -13,6 +13,27 @@ The live version of the documentation is synced with [live-docs repository](http
 
 - Try to follow [Google developer documentation style guide](https://developers.google.com/style).
 
+**Working with templates and auto-generated pieces:**
+
+###### Files that shouldn't be edited directly (follow described flow instead): 
+- cns-smart-contracts.md:
+	1. Edit *templates/cns-smart-contracts-template.md*
+	2. Run `render:cns-contracts`
+
+###### Auto-generated smart-contract address tables:
+
+Contract addresses are pulled from [network-config.json](https://github.com/unstoppabledomains/dot-crypto/blob/master/src/network-config/network-config.json "network-config.json") contained in [dot-crypto](https://github.com/unstoppabledomains/dot-crypto).
+
+- To fetch new config run `render:cns-contracts`. 
+- If you need to add a new Contract:
+
+	1. Create temlate for the target doc with the pattern of ``<target-doc-name>-temaplte.md`` and put it in a */template* folder.
+	2. Inside of a template insert `#include "templates/contracts/<YourContractName>.md"` where you want to add your table.
+	3. Generate tables & render the template using the following commands:
+		- `render:cns-contracts` for *cns-smart-contracts.md*
+		- Or define your own for other files.
```

Comment:
I'd probably not mention that and focus this readme just on the contract list to make it less confusing.

Reply:
Removed parts that are out of scope.